library/psa_crypto_slot_management.c

```diff
@@ -247,25 +247,42 @@ static psa_status_t psa_load_persistent_key_into_slot( psa_key_slot_t *slot )
     if( status != PSA_SUCCESS )
         goto exit;
 
-#if defined(MBEDTLS_PSA_CRYPTO_SE_C)
     if( psa_key_lifetime_is_external( slot->attr.lifetime ) )
     {
-        psa_se_key_data_storage_t *data;
-        if( key_data_length != sizeof( *data ) )
+#if defined(MBEDTLS_PSA_CRYPTO_SE_C)
+        const psa_drv_se_t *drv;
+        psa_drv_se_context_t *drv_context;
+        if( psa_get_se_driver( slot->attr.lifetime, &drv, &drv_context ) )
```

Comment:
Why not just changing the if( psa_key_lifetime_is_external( slot->attr.lifetime ) ) current check with that check ? That way it seems you don't have to duplicate the call to psa_copy_key_material_into_slot().

Reply:
Done.